app-android/src/main/java/com/ustadmobile/port/android/view/CourseTerminologyEditFragment.kt

```diff
@@ -93,4 +104,81 @@ class CourseTerminologyEditFragment: UstadEditFragment<CourseTerminology>(), Cou
             terminologyEntryAdapter?.submitList(value)
             terminologyEntryAdapter?.notifyDataSetChanged()
         }
+}
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+private fun CourseTerminologyEditScreen(
+    uiState: CourseTerminologyEditUiState = CourseTerminologyEditUiState(),
+    onTerminologyTermChanged: (TerminologyEntry?) -> Unit = {},
+    onCtTitleChanged: (String?) -> Unit = {}
+) {
+    LazyColumn(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(16.dp)
+    )  {
+
+        item{
+            ListItem(
+                text = {
+                    UstadTextEditField(
+                        value = uiState.entity?.ctTitle ?: "",
+                        label = stringResource(id = R.string.name),
+                        error = uiState.titleError,
+                        enabled = uiState.fieldsEnabled,
+                        onValueChange = { onCtTitleChanged(it) },
+                    )
+                },
+                secondaryText = { Text(stringResource(id = R.string.your_words_for)) }
+            )
+        }
+
+        items(
+            items = uiState.terminologyTermList,
+            key = {terminologyTerm -> terminologyTerm.id}
+        ){ terminologyTerm ->
+            ListItem(
+                text = {
+                    UstadTextEditField(
+                        value = terminologyTerm.term ?: "",
+                        label = stringResource(id = terminologyTerm.messageId),
+                        error = terminologyTerm.errorMessage,
+                        enabled = uiState.fieldsEnabled,
+                        onValueChange = {
+                            onTerminologyTermChanged(terminologyTerm.copy(
+                                term = it
+                            ))
+                        },)
+                },
+            )
+        }
+    }
+}
+
+@Composable
+@Preview
+fun CourseTerminologyEditScreenPreview() {
+    val uiState = CourseTerminologyEditUiState(
+        terminologyTermList = listOf(
+            TerminologyEntry(
+                id = "1",
+                term = "First",
+                messageId = R.string.message
```

Comment:
This should use MessageID e.g.
messageID = MessageID.teacher , MessageID.student, etc.